module,execution_time (sec),success,timestamp,error,error_short
cowidev.vax.batch.argentina,15.62,True,2023-05-28T07:00:21,,
cowidev.vax.batch.australia,,,2023-05-28T07:00:06,,
cowidev.vax.batch.austria_old,,,2023-05-28T07:00:06,,
cowidev.vax.batch.belgium,6.36,True,2023-05-28T07:00:12,,
cowidev.vax.batch.bolivia,4.05,True,2023-05-28T07:00:10,,
cowidev.vax.batch.canada,8.1,False,2023-05-28T07:02:01,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 282, in main
    Canada().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 270, in export
    df = df.pipe(self.pipeline, df_man)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 250, in pipeline
    .pipe(self.pipe_make_monotonic)[
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 230, in pipe_make_monotonic
    df = df.pipe(self.make_monotonic, max_removed_rows=self.max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/utils/base.py"", line 120, in make_monotonic
    return mkm(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 23, in make_monotonic
    return make_monotonic_new(df, column_date, column_metrics, max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 96, in make_monotonic_new
    raise Exception(
Exception: 1 chunks of lengths 36 would be NaNed for metric people_vaccinated. That is more than maximum allowed (22) by make_monotonic() - check the data. Check dates ['2023-01-26']. Location(s) affected: {'Canada'}
",1 chunks of lengths 36 would be NaNed for metric people_vaccinated. That is more than maximum allowed (22) by make_monotonic() - check the data. Check dates ['2023-01-26']. Location(s) affected: {'Canada'}
cowidev.vax.batch.chile,1.06,True,2023-05-28T07:00:07,,
cowidev.vax.batch.czechia,22.21,True,2023-05-28T07:00:28,,
cowidev.vax.batch.denmark,,,2023-05-28T07:00:06,,
cowidev.vax.batch.ecdc,94.68,True,2023-05-28T07:01:40,,
cowidev.vax.batch.ecuador,,,2023-05-28T07:00:06,,
cowidev.vax.batch.estonia,1.17,True,2023-05-28T07:00:07,,
cowidev.vax.batch.france,5.42,True,2023-05-28T07:00:11,,
cowidev.vax.batch.germany,0.25,True,2023-05-28T07:00:06,,
cowidev.vax.batch.greece,7.39,True,2023-05-28T07:00:13,,
cowidev.vax.batch.hong_kong,22.31,True,2023-05-28T07:00:28,,
cowidev.vax.batch.indonesia,,,2023-05-28T07:00:06,,
cowidev.vax.batch.ireland,3.69,True,2023-05-28T07:00:09,,
cowidev.vax.batch.israel,4.37,True,2023-05-28T07:00:10,,
cowidev.vax.batch.italy,24.44,True,2023-05-28T07:00:30,,
cowidev.vax.batch.japan,9.42,False,2023-05-28T07:02:11,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 375, in main
    Japan().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 359, in export
    df = self.read().pipe(self.pipeline_base)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 71, in read
    df_latest = self.read_latest().pipe(self.pipe_read_latest)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 100, in read_latest
    self._read_latest_booster(),
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 211, in _read_latest_booster
    df = self._read_xlsx(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 233, in _read_xlsx
    raise ValueError(f""Unknown sheets: {sheets_unknown}"")
ValueError: Unknown sheets: {'5.8以降接種_一般接種（６回目）', '６回目_総接種回数', '６回目_一般接種回数', '5.8以降接種_総接種回数', '5.8以降接種_一般接種（３回目）', '５回目_一般接種回数', '5.8以降接種_一般接種（５回目）', '5.8以降接種_一般接種（４回目）', '５回目_総接種回数'}
","Unknown sheets: {'5.8以降接種_一般接種（６回目）', '６回目_総接種回数', '６回目_一般接種回数', '5.8以降接種_総接種回数', '5.8以降接種_一般接種（３回目）', '５回目_一般接種回数', '5.8以降接種_一般接種（５回目）', '5.8以降接種_一般接種（４回目）', '５回目_総接種回数'}"
cowidev.vax.batch.jersey,4.34,True,2023-05-28T07:00:10,,
cowidev.vax.batch.latvia,61.18,True,2023-05-28T07:01:07,,
cowidev.vax.batch.lithuania,4.7,True,2023-05-28T07:00:10,,
cowidev.vax.batch.luxembourg,3.32,True,2023-05-28T07:00:09,,
cowidev.vax.batch.malaysia,0.97,True,2023-05-28T07:00:07,,
cowidev.vax.batch.malta,1.14,True,2023-05-28T07:00:07,,
cowidev.vax.batch.netherlands,,,2023-05-28T07:00:06,,
cowidev.vax.batch.new_zealand,14.05,True,2023-05-28T07:00:20,,
cowidev.vax.batch.norway,0.54,True,2023-05-28T07:00:06,,
cowidev.vax.batch.peru,3.77,True,2023-05-28T07:00:09,,
cowidev.vax.batch.portugal,,,2023-05-28T07:00:06,,
cowidev.vax.batch.romania,42.0,True,2023-05-28T07:00:48,,
cowidev.vax.batch.saudi_arabia,5.69,True,2023-05-28T07:00:11,,
cowidev.vax.batch.singapore,3.98,True,2023-05-28T07:00:10,,
cowidev.vax.batch.slovakia,48.03,True,2023-05-28T07:00:54,,
cowidev.vax.batch.slovenia,3.68,True,2023-05-28T07:00:09,,
cowidev.vax.batch.south_korea,88.76,True,2023-05-28T07:01:34,,
cowidev.vax.batch.spc,54.47,True,2023-05-28T07:01:00,,
cowidev.vax.batch.sweden,25.2,True,2023-05-28T07:00:31,,
cowidev.vax.batch.switzerland,88.21,True,2023-05-28T07:01:35,,
cowidev.vax.batch.trinidad_and_tobago,4.1,True,2023-05-28T07:00:11,,
cowidev.vax.batch.ukraine,,,2023-05-28T07:00:07,,
cowidev.vax.batch.united_kingdom,23.51,True,2023-05-28T07:00:30,,
cowidev.vax.batch.united_states,14.58,True,2023-05-28T07:00:21,,
cowidev.vax.batch.uruguay,2.18,True,2023-05-28T07:00:11,,
cowidev.vax.batch.zimbabwe,0.93,True,2023-05-28T07:00:10,,
cowidev.vax.incremental.africacdc,,,2023-05-28T07:00:09,,
cowidev.vax.incremental.antigua_barbuda,,,2023-05-28T07:00:09,,
cowidev.vax.incremental.aruba,1.06,True,2023-05-28T07:00:10,,
cowidev.vax.incremental.austria,,,2023-05-28T07:00:09,,
cowidev.vax.incremental.azerbaijan,,,2023-05-28T07:00:09,,
cowidev.vax.incremental.bahrain,,,2023-05-28T07:00:09,,
cowidev.vax.incremental.bangladesh,61.3,True,2023-05-28T07:01:11,,
cowidev.vax.incremental.barbados,,,2023-05-28T07:00:10,,
cowidev.vax.incremental.brazil,0.41,True,2023-05-28T07:00:10,,
cowidev.vax.incremental.bulgaria,1.82,True,2023-05-28T07:00:12,,
cowidev.vax.incremental.china,19.81,True,2023-05-28T07:00:30,,
cowidev.vax.incremental.costa_rica,,,2023-05-28T07:00:10,,
cowidev.vax.incremental.croatia,0.79,True,2023-05-28T07:00:11,,
cowidev.vax.incremental.cuba,24.44,True,2023-05-28T07:00:35,,
cowidev.vax.incremental.curacao,,,2023-05-28T07:00:10,,
cowidev.vax.incremental.cyprus,1.15,True,2023-05-28T07:00:11,,
cowidev.vax.incremental.dominican_republic,13.24,False,2023-05-28T07:02:24,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 112, in main
    DominicanRepublic().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 68, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 26, in read
    metrics = self._parse_metrics(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 35, in _parse_metrics
    metric_candidates = _find_potential_metrics(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 91, in _find_potential_metrics
    h3 = _find_h3(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 81, in _find_h3
    raise ValueError(""More than one element found!"")
ValueError: More than one element found!
",More than one element found!
cowidev.vax.incremental.el_salvador,5.25,True,2023-05-28T07:00:16,,
cowidev.vax.incremental.faeroe_islands,0.77,True,2023-05-28T07:00:12,,
cowidev.vax.incremental.fiji,0.01,True,2023-05-28T07:00:11,,
cowidev.vax.incremental.finland,0.27,True,2023-05-28T07:00:11,,
cowidev.vax.incremental.gabon,,,2023-05-28T07:00:11,,
cowidev.vax.incremental.georgia,2.18,True,2023-05-28T07:00:13,,
cowidev.vax.incremental.greenland,,,2023-05-28T07:00:11,,
cowidev.vax.incremental.guatemala,,,2023-05-28T07:00:11,,
cowidev.vax.incremental.guernsey,,,2023-05-28T07:00:11,,
cowidev.vax.incremental.hungary,87.93,True,2023-05-28T07:01:39,,
cowidev.vax.incremental.iceland,0.77,True,2023-05-28T07:00:12,,
cowidev.vax.incremental.india,1.71,True,2023-05-28T07:00:13,,
cowidev.vax.incremental.iran,,,2023-05-28T07:00:11,,
cowidev.vax.incremental.isle_of_man,2.73,True,2023-05-28T07:00:14,,
cowidev.vax.incremental.jamaica,16.89,True,2023-05-28T07:00:28,,
cowidev.vax.incremental.kazakhstan,12.01,True,2023-05-28T07:00:24,,
cowidev.vax.incremental.kosovo,,,2023-05-28T07:00:12,,
cowidev.vax.incremental.kyrgyzstan,1.68,True,2023-05-28T07:00:14,,
cowidev.vax.incremental.laos,2.06,True,2023-05-28T07:02:26,,
cowidev.vax.incremental.macao,8.9,False,2023-05-28T07:02:35,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 90, in main
    Macao().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 75, in export
    data = self.read()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 23, in read
    iframe_url = self._get_iframe_url(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 31, in _get_iframe_url
    elem = driver.find_element_by_id(""ICovid19Monitor"")
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 360, in find_element_by_id
    return self.find_element(by=By.ID, value=id_)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 976, in find_element
    return self.execute(Command.FIND_ELEMENT, {
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 321, in execute
    self.error_handler.check_response(response)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py"", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {""method"":""css selector"",""selector"":""[id=""ICovid19Monitor""]""}
  (Session info: headless chrome=113.0.5672.126)

","Message: no such element: Unable to locate element: {""method"":""css selector"",""selector"":""[id=""ICovid19Monitor""]""}
  (Session info: headless chrome=113.0.5672.126)
"
cowidev.vax.incremental.moldova,,,2023-05-28T07:00:13,,
cowidev.vax.incremental.monaco,95.21,True,2023-05-28T07:01:48,,
cowidev.vax.incremental.mongolia,1.56,True,2023-05-28T07:00:15,,
cowidev.vax.incremental.montenegro,0.84,True,2023-05-28T07:00:14,,
cowidev.vax.incremental.morocco,,,2023-05-28T07:00:14,,
cowidev.vax.incremental.myanmar,,,2023-05-28T07:00:14,,
cowidev.vax.incremental.nepal,,,2023-05-28T07:00:14,,
cowidev.vax.incremental.north_macedonia,,,2023-05-28T07:00:14,,
cowidev.vax.incremental.northern_cyprus,13.05,True,2023-05-28T07:00:27,,
cowidev.vax.incremental.paho,90.12,False,2023-05-28T07:04:05,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 195, in main
    PAHO().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 190, in export
    df = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 167, in pipeline
    df.pipe(self.pipe_check_columns)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 122, in pipe_check_columns
    raise ValueError(f""Missing column fields: {columns_missing}. Present columns are: {df.columns}"")
ValueError: Missing column fields: {'Single dose', 'Total doses', '2nd additional dose', 'First dose', 'Country code', 'Second dose', '1st additional dose', 'Country/ Territory'}. Present columns are: Index(['date,location,total_vaccinations,people_vaccinated,people_fully_vaccinated,total_boosters,new_vaccinations,new_vaccinations_interpolated,new_people_vaccinated_interpolated,new_vaccinations_smoothed,new_people_vaccinated_smoothed', 'date'], dtype='object')
","Missing column fields: {'Single dose', 'Total doses', '2nd additional dose', 'First dose', 'Country code', 'Second dose', '1st additional dose', 'Country/ Territory'}. Present columns are: Index(['date,location,total_vaccinations,people_vaccinated,people_fully_vaccinated,total_boosters,new_vaccinations,new_vaccinations_interpolated,new_people_vaccinated_interpolated,new_vaccinations_smoothed,new_people_vaccinated_smoothed', 'date'], dtype='object')"
cowidev.vax.incremental.pakistan,,,2023-05-28T07:00:14,,
cowidev.vax.incremental.philippines,,,2023-05-28T07:00:14,,
cowidev.vax.incremental.poland,2.33,True,2023-05-28T07:00:17,,
cowidev.vax.incremental.qatar,,,2023-05-28T07:00:15,,
cowidev.vax.incremental.russia,36.43,True,2023-05-28T07:00:51,,
cowidev.vax.incremental.saint_lucia,,,2023-05-28T07:00:16,,
cowidev.vax.incremental.serbia,14.41,True,2023-05-28T07:00:30,,
cowidev.vax.incremental.south_africa,,,2023-05-28T07:00:17,,
cowidev.vax.incremental.spain,14.07,True,2023-05-28T07:00:31,,
cowidev.vax.incremental.sri_lanka,,,2023-05-28T07:00:20,,
cowidev.vax.incremental.suriname,15.5,True,2023-05-28T07:00:35,,
cowidev.vax.incremental.taiwan,11.53,False,2023-05-28T07:04:17,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 184, in main
    Taiwan().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 170, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 37, in read
    df = self._parse_table(url_pdf)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 76, in _parse_table
    raise ValueError(f""There are some unknown columns: {cols}"")
ValueError: There are some unknown columns: Index(['廠牌', '劑次', '4 月接種人次', '累計至 4 月接種人次'], dtype='object')
","There are some unknown columns: Index(['廠牌', '劑次', '4 月接種人次', '累計至 4 月接種人次'], dtype='object')"
cowidev.vax.incremental.thailand,8.29,True,2023-05-28T07:00:30,,
cowidev.vax.incremental.turkey,,,2023-05-28T07:00:23,,
cowidev.vax.incremental.united_arab_emirates,,,2023-05-28T07:00:23,,
cowidev.vax.incremental.vietnam,,,2023-05-28T07:00:23,,
cowidev.vax.incremental.who,44.77,True,2023-05-28T07:01:08,,
cowidev.vax.incremental.zambia,2.87,True,2023-05-28T07:00:27,,
